### PR TITLE
Ensure we reset OutputtedFilenameList in profiling when a workspace is restored

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This patch ensures that if we load a saved workspace, we correctly output filenames. This is only required when restoring a workspace where coverage/profile was active, with `--prof` or `--cover`.

The workspace restores the variable `OutputtedFilenameList`, which represents files we have already outputted. We want to clear that file, so our new profile includes the filenames.

This is a pain to add a test for, but easy to eyeball -- looking at profiles generated they were previously missing filenames, now they aren't.

### CODE REVIEW

File: src/profile.c [hunk 1]
```diff
@@ -1005,6 +1005,15 @@ static Int InitKernel (
     return 0;
 }
 
+static Int PostRestore ( StructInitInfo * module )
+{
+    /* When we restore a workspace, we start a new profile. This is the
+     * only part of the profile which is stored in the GAP memory space,
+     * so we want to clear it
```
Comment (fingolfin) [inline src/profile.c]: We "want" to clear it? Not "need" to clear it?

File: src/profile.c [hunk 2]
```diff
@@ -1005,6 +1005,15 @@ static Int InitKernel (
     return 0;
 }
 
+static Int PostRestore ( StructInitInfo * module )
+{
+    /* When we restore a workspace, we start a new profile. This is the
```
Comment (fingolfin) [inline src/profile.c]: I was mildly confused when reading "This" here, until I realized that it referred to the text after the comment.  Perhaps rephrase this as "The OutputtedFilenameList is ..." or "The list of outputted filenames is" ?

Comment (ChrisJefferson) [inline src/profile.c]: Tweaked

File: src/profile.c (hunk 1 above)
Comment (ChrisJefferson) [inline src/profile.c]: I'm happy to change to 'need'

**Comment by ChrisJefferson** [comment]
I have no idea why it looks like this is trashing coverage so badly.

**Comment by ChrisJefferson** [comment]
I have no idea why it looks like this is trashing coverage so badly.
